samples/client/petstore/typescript-fetch/builds/enum/models/StringOneOf.ts

```diff
@@ -0,0 +1,51 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * Enum test
+ * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+ *
+ * The version of the OpenAPI document: 1.0.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import {
+    StringEnum,
+    StringEnumTwo,
+    StringEnumFromJSONTyped,
+    StringEnumToJSON,
+    StringEnumTwoFromJSONTyped,
+    StringEnumTwoToJSON,
+} from './';
+
+/**
+ * @type StringOneOf
+ * 
+ * @export
+ */
+export type StringOneOf = StringEnum | StringEnumTwo;
+
+export function StringOneOfFromJSON(json: any): StringOneOf {
+    return StringOneOfFromJSONTyped(json, false);
+}
+
+export function StringOneOfFromJSONTyped(json: any, ignoreDiscriminator: boolean): StringOneOf {
+    if ((json === undefined) || (json === null)) {
+        return json;
+    }
+    return StringEnumFromJSONTyped(json, true) || StringEnumTwoFromJSONTyped(json, true);
+}
+
+export function StringOneOfToJSON(value?: StringOneOf | null): any {
+    if (value === undefined) {
+        return undefined;
+    }
+    if (value === null) {
+        return null;
+    }
+    return StringEnumToJSON(value as StringEnum) || StringEnumTwoToJSON(value as StringEnumTwo);
```

Comment:
the second part after `||` will never work.
It's not a big deal for string enums, which don't really convert values and only do type assertion. However, for more complex data types it's a potential problem - right?

Reply:
It didn't cross my mind, our docs only use string enums. I test it out tomorrow and see!